indico/htdocs/js/indico/jquery/rulelistwidget.js

```diff
@@ -0,0 +1,213 @@
+                            $this.show();
+                        } else {
+                            $this.hide();
+                            delete ruleContext.rule[conditionName];
+                        }
+                    });
+                }
+            });
+        },
+
+        _setConditionOption: function(ruleContext, conditionName, value) {
+            var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + conditionName + ']');
+            var $select = $condition.find('select');
+
+            if (value === undefined) {
+                value = this.options.conditionChoices[conditionName].options[0][0];
+            }
+
+            // will implicitly call _optionSelected(...)
+            $select.val(_optionValue(value));
+        },
+
+        _checkPlaceholder: function() {
+            var $placeholder = this.options.containerElement.find('.no-rules-placeholder');
+            if (this.ruleData.length) {
+                $placeholder.hide();
+            } else {
+                $placeholder.show();
+            }
+        },
+
+        _addRule: function($ruleList, rule) {
+            var self = this;
+            var ruleContext = {
+                $element: $('<li>', {class: 'rule'}),
+                rule: rule
+            };
+            var $removeButton = $('<a>', {class: 'i-button-icon icon-remove rule-remove-button'});
+
+            ruleContext.$element.append(
+                _.map(this.conditionSpec, function(condition) {
+                    var $html = self._renderRuleCondition(ruleContext, condition);
+                    return $html;
+                }));
+            ruleContext.$element.append($removeButton).appendTo($ruleList);
+            ruleContext.$element.data('ruleContext', ruleContext);
+
+            this.conditionSpec.forEach(function(condition) {
+                self._setConditionOption(ruleContext, condition.value, rule[condition.value]);
+            });
+            this._checkConditionIncompatibility(ruleContext);
+            this._checkPlaceholder();
+            this.element.trigger('change');
+        },
+
+        _removeRule: function($elem) {
+            var rule = $elem.data('ruleContext').rule;
+            this.ruleData = _.without(this.ruleData, _.find(this.ruleData, _.partial(_.isEqual, rule)));
+            $elem.remove();
+            this._checkPlaceholder();
+            this.element.trigger('change');
+        },
+
+        _addNewRule: function($ruleList) {
+            var firstChoice = this.options.conditionTypes[0];
+            var newRule = {};
+
+            newRule[firstChoice] = _listValue(this.options.conditionChoices[firstChoice].options[0][0]);
+            this.ruleData.push(newRule);
+            this._addRule($ruleList, newRule);
+        },
+
+        _create: function() {
+            var self = this;
+            var $container = this.options.containerElement;
```

Comment:
if you already use `self` in a method i'd use it all the time to make it clearer it's the same thing